libraries/tock-register-interface/src/interfaces.rs

```diff
@@ -230,17 +230,30 @@ pub trait Readable {
         field.is_set(self.get())
     }
 
+    /// Check if any bits corresponding to the mask in the passed `FieldValue` are set.
+    /// This function is identical to `is_set()` but operates on a `FieldValue` rather
+    /// than a `Field`, allowing for checking if any bits are set across multiple,
+    /// non-contiguous portions of a bitfield.
     #[inline]
-    /// Check if any specified parts of a field match
-    fn matches_any(&self, field: FieldValue<Self::T, Self::R>) -> bool {
-        field.matches_any(self.get())
+    fn any_matching_bits_set(&self, field: FieldValue<Self::T, Self::R>) -> bool {
+        field.any_matching_bits_set(self.get())
     }
 
     #[inline]
     /// Check if all specified parts of a field match
     fn matches_all(&self, field: FieldValue<Self::T, Self::R>) -> bool {
         field.matches_all(self.get())
```

Comment:
It seems like `matches_all()` needs to change as well. This is now `all_matching_bits_set()`, right?

Reply:
`matches_all()` is actually already correct -- since it was already checking for an exact match across the entire passed `FieldValue`, it works correctly for multi-bit fields and for matching on cleared bits as well as set bits. 

For convenience here are the two implementations:
```rust
    /// Check if any of the bits covered by the mask for this
    /// `FieldValue` are set in the passed value
    #[inline]
    pub fn any_matching_bits_set(&self, val: T) -> bool {
        val & self.mask != T::zero()
    }

    /// Check if all specified parts of a field match
    #[inline]
    pub fn matches_all(&self, val: T) -> bool {
        val & self.mask == self.value
    }
```

`matches_all()` actually looks at `self.value()` and makes sure that the passed value, when masked, matches it. So the problem in #3311 does not apply.

Reply:
Ok good point. Seems like if there is only one FieldValue, then it should be `.matches()`.

Reply:
Yeah, that makes sense to me, but I don't think the current name is really *wrong* ... not sure how to balance that with a desire not to modify the API of tock-registers more than needed, especially since it is the one library with non-Tock downstream users